docs/v2-migration-guide.rst

```diff
@@ -179,6 +179,78 @@ bug fixes will be shipped to the 1.26.x release stream**.
 If your organization relies on urllib3 and is interested in continuing support you can learn
 more about the `Tidelift Subscription for Enterprise <https://tidelift.com/subscription/pkg/pypi-urllib3?utm_source=pypi-urllib3&utm_medium=referral&utm_campaign=docs>`_.
 
+**🤔 Common upgrading issues**
+-------------------------------
+
+The following sections explain common issues and their remediation. If all else fails, you should
+pin urllib3 to 1.26.x using ``urllib3<2`` in your requirements. For more details on the recommended
+way to handle your dependencies in general, see `Semantic Versioning Will Not Save You
+<https://hynek.me/articles/semver-will-not-save-you/>`_. The second half even uses urllib3 2.0 as an
+example.
+
+ssl module is compiled with OpenSSL 1.0.2.k-fips
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: text
+
+  ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with OpenSSL 1.0.2k-fips  26 Jan 2017.
+  See: https://github.com/urllib3/urllib3/issues/2168
+
+Remediation depends on your system.
+
+- **AWS Lambda**: Upgrade to the Python 3.9 or 3.10 runtimes that come OpenSSL 1.1.1.
```

Comment:
Is this accurate for our current understanding? Perhaps we should recommend folks use Python 3.10?

Reply:
Yes, it's accurate, but I changed this to only recommend 3.10 for simplicity: [`8f87259` (#3018)](https://github.com/urllib3/urllib3/pull/3018/commits/8f87259b949028efd65ccc10b5260101921167a8)